analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -250,6 +250,7 @@ class SpdxDocumentFile(
     analyzerConfig: AnalyzerConfiguration,
     repoConfig: RepositoryConfiguration
 ) : PackageManager(managerName, analysisRoot, analyzerConfig, repoConfig) {
+    private val spdxDocumentForFile = mutableMapOf<File, SpdxDocument>()
```

Comment:
As this increases memory consumption could you please indicate why this is needed?

Reply:
As the commit message states: To avoid parsing the same file twice. So we basically trade lower CPU usage for higher memory usage.

Reply:
Actually, this change probably even *decreases* memory usage. Because previously, the same file has been parsed twice, and probably both copies were kept in memory, as they are parsed very quickly after each other, before garbage collection kicks in.

Reply:
> To avoid parsing the same file twice.

That's clear, I was more interested in how big the performance issue is.
It also adds the hidden assumption that package manager is destroyed by the caller to free memory.
Anyhow, I'm ok with the change.